### PR TITLE
Add xds-enabled server creation details for Core and C++

### PR DESCRIPTION
**Pull request by yashykt**
I've updated the A36-xds-for-servers.md doc with API details for Core and C++. Please let me know if a new gRFC would be preferred.

### CODE REVIEW

File: A36-xds-for-servers.md
Comment (markdroth) [inline A36-xds-for-servers.md:343]: s/dealt with the/dealt with via the/

File: A36-xds-for-servers.md
Comment (markdroth) [inline A36-xds-for-servers.md:335]: I think we should say this a little more strongly: it means that it is not supported to use an ephemeral port with and xDS-enabled gRPC server.

File: A36-xds-for-servers.md
Comment (markdroth) [inline A36-xds-for-servers.md:388]: Same as above: We should explicitly state that ephemeral ports are not supported via `XdsServerBuilder`.

File: A36-xds-for-servers.md
Comment (markdroth) [inline A36-xds-for-servers.md:391]: The second paragraph of this section can now be removed (the one starting with "C will expose an API for wrapped languages to utilize").

**Review by markdroth — COMMENTED** [review]
Thanks for doing this!  Just a few minor comments.

File: A36-xds-for-servers.md
Comment (ejona86) [inline A36-xds-for-servers.md:345]: I thought we decided that the wildcard port was chosen by listen(). So this isn't applicable. I removed the language about splitting bind+listen long before merging the gRFC. The choices are "not listen()" or "accept()+close()".

File: A36-xds-for-servers.md
Comment (yashykt) [reply A36-xds-for-servers.md:335]: The reason for the not-so-strong language is to allow for flexibility to add support for wildcard ports, unless we want to say that we are never going to consider doing that.

File: A36-xds-for-servers.md
Comment (yashykt) [reply A36-xds-for-servers.md:345]: This is the relevant comment from @markdroth in the original PR
```
I think it's worth pointing out that calling listen() before the xDS resource is obtained is actually worse for clients, because even without calling accept(), it will still cause timeout delays before pick_first can fail over to the next address in the list. So I think implementations should prefer to defer calling listen() until after the xDS resource is obtained. That's what we're planning to do in C-core.

Also, the language here seems to imply that bind() should be done immediately even if listen() is delayed. Is there actually any reason for that? Based on what @yashykt and I are looking at in C-core, it seems like the only case where it would matter that bind() happens before we get the xDS resource is a server using an ephemeral port. And that seems like such a niche case that I think I'd be okay saying that we just don't support it for the xDS-enabled server API. (To be clear, it does seem better to call bind() earlier if possible, so that we can support ephemeral ports, but I think it's a nice-to-have, not a hard requirement. In C-core, it looks like that may be hard to de-couple, so at least initially we will probably start by deferring both bind() and listen() until after we have the xDS resource. We can revisit that later if someone actually wants to use ephemeral ports with xDS.)
```
From this I'm interpreting that splitting "bind()" and "listen()" is still a possible strategy.

File: A36-xds-for-servers.md
Comment (yashykt) [reply A36-xds-for-servers.md:321]: These channel args are used for the creation of the xds client. At present, we only seem to use this to pass in ```GRPC_ARG_TEST_ONLY_DO_NOT_USE_IN_PROD_XDS_BOOTSTRAP_CONFIG```. I'm not sure if we would need this for anything else in the future. @markdroth

File: A36-xds-for-servers.md
Comment (ejona86) [reply A36-xds-for-servers.md:345]: There was a lot more discussion than that[1][2][3], but in the end all the reference to bind() was removed in favor of focusing on listen() and "XdsServer will not support ephemeral ports as part of this gRFC but may be enhanced in the future."

I guess this text is fine as-is. It initially seemed like a lot about a topic that wasn't part of the gRFC, but it is really only one mention of splitting bind()+listen(), but another mention to bind() to make that make sense. That's fair. Let's keep it as-is.

1. https://github.com/grpc/proposal/pull/214#issuecomment-749280531
2. https://github.com/grpc/proposal/pull/214#discussion_r552172048
3. after other discussions, https://github.com/grpc/proposal/pull/214/commits/a933f33729a327fc153274b71fdd6a41b340b25b

File: A36-xds-for-servers.md
Comment (markdroth) [reply A36-xds-for-servers.md:302]: +1 -- this should say "C-core".

File: A36-xds-for-servers.md
Comment (markdroth) [reply A36-xds-for-servers.md:321]: That arg is the reason we plumbed the channel args in there, but in principle they could be used for other things later.  They will just be whatever args were passed to the server.

File: A36-xds-for-servers.md
Comment (markdroth) [reply A36-xds-for-servers.md:335]: I think the current language is so fuzzy that it really doesn't tell the reader anything useful.  If the concern is that we want to be able to change this later, then let's just say something like "Note that we will initially not support ephemeral ports on an xDS-enabled server, but we might add that later."

**Review by markdroth — APPROVED** [review]
Looks great!

**Review by ejona86 — APPROVED** [review]

**Comment by yashykt** [comment]
Thanks for the reviews!
I'll merge this next Friday assuming there are no objections on the API.

**Comment by yashykt** [comment]
Don't see any objections on the discussion thread. Merging this. Thanks for reviewing!

**Pull request merged**
